examples/2Dturbulence_multigrid/data/BOUT.inp

```diff
@@ -10,9 +10,6 @@ zmin = 0
 zmax = 15.91549431       # 15.915*2pi = 100.  z is in fractions of 2pi
 myg = 0                  # No need for Y communications
 
-[ddx]
-#upwind = w3             # u1 (default), u4, w3 
-
```

Comment:
Is removing this a bug? The section has no options, but does have a commented-out option, and we keep commented out options elsewhere, e.g. just below in `[ddz]`. This happens a fair few more times below...

Reply:
This particular case was me starting to manually fix this sort of issue before I realised the scope of the problem. I guess I didn't undo this change before committing, sorry!

The overall problem is that the logic in the python `BoutOptions` is that comments are either: inline and attached to the key on the same line; or out-of-line and attached to the _next_ option or section.

So in this case, `#upwind = w3   (...)          # u1 (default), u4, w3` is a comment attached to the `[ddz]` section, so it would normally stay, if I hadn't deleted it manually.

Same thing in your second comment, with `# max_mode = 16  # Maximum N to solve for` actually part of a larger comment block attached to the `[fft]` section.

In your third comment, the comments are "hanging": commented out stuff at the bottom of the file isn't attached to anything and so gets ignored during the parsing stage and so dropped when we write the file back out.

I don't particular want to change `BoutOptions` to try and deal with all these cases, as it's in general hard to tell the difference between a commented out option and a "real" comment on the next section/option. Too many corner cases!

I think it's not _terrible_ to leave these orphaned commented-out options, but I'm happy to go through more thoroughly and fix them if you think it would be better.

Reply:
The input files in the tests and examples are a bit like documentation - people might want to copy them or use them for reference, so I do think it'd be a good idea to tidy them up so there aren't weird things that don't really make sense in there.

Probably the best thing is to fully delete all the empty sections. Although I think my personal preference would be 'attach comments to the next option if there's an option, otherwise attach them to the previous section' so that everything between one section heading and the next 'belongs to' the section it's in*. That's probably a faff to implement though...

\* I realise there are some 'comments on the section' before section headings in the examples, so I'd just propose to consider that pattern non-canonical and ignore it, possibly tidying up by either deleting comments or moving them under the section heading.

Reply:
Very true, I completely agree. You can even see where we've copied input files for other examples :smile: 

I've been through and fixed the most egregious cases. There's still lots of diffs github didn't want to show me, and I really don't want to go through another 50 files locally!

The trouble with the comments is that people are remarkably inconsistent, and there are so many corner cases, so I'm not terribly keen on trying to capture all that. Even if we pick a canonical style (which is a Good Idea), we'd still need to do a massive amount of tidying up -- there's 150-200 input files!

At some point we should have a big tidy-up of the examples, and cull ones we think are no longer relevant.

I think this PR is probably ok for now though